### PR TITLE
Make 'compileOnly' extend 'compileOnlyApi'

### PR DESCRIPTION
**Pull request by jjohannes**
This is in line with `implementation` extends `api`.

With this, existing plugins, that rely on extending (and then resolving) compileOnly, automatically support the new compileOnlyApi.

### CODE REVIEW

**Comment by jjohannes** [comment]
@bot-gradle test QF

File: subprojects/plugins/src/main/java/org/gradle/api/plugins/internal/JvmPluginsHelper.java [hunk 1]
```diff
@@ -97,7 +97,7 @@ public static Configuration addApiToSourceSet(SourceSet sourceSet, Configuration
         Configuration implementationConfiguration = configurations.getByName(sourceSet.getImplementationConfigurationName());
         implementationConfiguration.extendsFrom(apiConfiguration);
 
-        Configuration compileClasspathConfiguration = configurations.getByName(sourceSet.getCompileClasspathConfigurationName());
+        Configuration compileClasspathConfiguration = configurations.getByName(sourceSet.getCompileOnlyConfigurationName());
```
Comment (ljacomet) [inline subprojects/plugins/src/main/java/org/gradle/api/plugins/internal/JvmPluginsHelper.java:100]: ⭕ Can you rename the variable as well?

**Review by ljacomet — CHANGES_REQUESTED** [review]
Looks good.

One comment about a variable name 😉

File: subprojects/plugins/src/main/java/org/gradle/api/plugins/internal/JvmPluginsHelper.java (hunk 1 above)
Comment (jjohannes) [reply subprojects/plugins/src/main/java/org/gradle/api/plugins/internal/JvmPluginsHelper.java]: Just saw that myself. Done.

**Comment by jjohannes** [comment]
@bot-gradle test QF

**Pull request merged**